src/ipc/ipc4/handler.c

```diff
@@ -823,6 +826,79 @@ static int ipc4_get_large_config_module_instance(struct ipc4_message_request *ip
 	return ret;
 }
 
+static int ipc4_set_vendor_config_module_instance(uint32_t module_id,
+						  uint32_t instance_id,
+						  bool init_block,
+						  bool final_block,
+						  uint32_t data_off_size,
+						  const char *data,
```

Comment:
I'd say we have mostly 3 options for this:
1. `void *` - because functions like `write(2)` use it, but arithmetics is difficult with it
2. `char *` - because `char` is simple, everybody knows, that it's 1 byte...
3. `uint8_t *` - because it's explicitly a pointer to a byte array

SOF uses all 3 options in different cases. In general I'd probably use (1), but since we already use them all... Feel free to make your choice, my comment was a proposal, not a hard requirement.

Reply:
Leaving this one as it is for now.